src/task_scl_eff_pot_hab.py

```diff
@@ -188,49 +165,72 @@ def density_to_patch_size(ft):
                 self.density_values["step_size_limits"]["min"],
                 self.density_values["step_size_limits"]["max"],
             )
-            .updateMask(hii_ecoregion)
+            .updateMask(potential_habitat)
         )
-        hii_grp = hii_ecoregion.addBands(
-            [
-                probability_calc,
-                hii_sum,
-                ee.Image.pixelArea().multiply(0.000001),
-                min_patch_size,
-                min_stepping_stone_size,
-            ]
-        ).reproject(crs="EPSG:4326", scale=self.scale)
-
-        polygonReducer = (
-            ee.Reducer.mode().unweighted()
-            .combine(ee.Reducer.mode().unweighted(), "hii_")
-            .combine(ee.Reducer.sum().unweighted(), "area_")
-            .combine(ee.Reducer.mode().unweighted(), "patch_")
-            .combine(ee.Reducer.mode().unweighted(), "stp_stn_")
+
+        potential_core = (
+            ee.Image(0).where(potential_habitat.gte(min_patch_size), 1).selfMask()
         )
-        hii_grp_poly = hii_grp.reduceToVectors(
-            reducer=polygonReducer,
-            geometry=ee.Geometry.MultiPolygon(self.aoi).bounds(),
-            crs=self.crs,
-            scale=self.scale,
-            maxPixels=self.ee_max_pixels,
-            geometryInNativeProjection=True,
-        ).select(
-            ["area_sum", "hii_mode", "label", "mode", "patch_mode", "stp_stn_mode"],
-            [
-                "patch_area",
-                "hii",
-                "label",
-                "prob",
-                "min_patch_area",
-                "min_stp_stn_area",
-            ],
+
+        potential_stepping_stone = (
+            ee.Image(0)
+            .where(
+                potential_habitat.lt(min_patch_size).And(
+                    potential_habitat.gte(min_stepping_stone_size)
+                ),
+                1,
+            )
+            .selfMask()
         )
 
-        scl_eff_pot_hab = "{}/geographies/Sumatra/scl_poly/{}/scl_eff_pot_hab".format(
-            self.species, self.taskdate
+        potential_core = (
+            dilate(potential_core, 2)
```

Comment:
Should the `2` argument be a var so it's clear what it is?

Reply:
done. now called self.thresholds['connectity_distance']. 
the function takes a value in pixels.  but perhaps the input should be km and we have a function to convert?  Currently running everything at a 1000 km scale is clean in this way, but if we may want to make this flexible if we change scales?